src/metabase/models/native_query_snippet.clj

```diff
@@ -84,25 +85,30 @@
   [_model-name _opts snippet]
   (-> (serdes/extract-one-basics "NativeQuerySnippet" snippet)
       (update :creator_id serdes/*export-user*)
-      (update :collection_id #(when % (serdes/*export-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*export-fk* % 'Collection))))
 
 (defmethod serdes/load-xform "NativeQuerySnippet" [snippet]
   (-> snippet
       serdes/load-xform-basics
       (update :creator_id serdes/*import-user*)
-      (update :collection_id #(when % (serdes/*import-fk* % 'Collection)))))
+      (m/update-existing :collection_id #(serdes/*import-fk* % 'Collection))))
 
 (defmethod serdes/dependencies "NativeQuerySnippet"
   [{:keys [collection_id]}]
   (if collection_id
     [[{:model "Collection" :id collection_id}]]
     []))
 
-(defmethod serdes/storage-path "NativeQuerySnippet" [snippet ctx]
-  ;; Intended path here is ["snippets" "nested" "collections" "snippet_eid_and_slug"]
-  ;; We just the default path, then pull it apart.
-  ;; The default is ["collections" "nested" collections" "nativequerysnippets" "base_name"]
-  (let [basis  (serdes/storage-default-collection-path snippet ctx)
-        file   (last basis)
-        colls  (->> basis rest (drop-last 2))] ; Drops the "collections" at the start, and the last two.
-    (concat ["snippets"] colls [file])))
+(defmethod serdes/storage-path "NativeQuerySnippet" [snippet _ctx]
+  (let [{:keys [id label]} (-> snippet serdes/path last)]
+    ["snippets" (serdes/storage-leaf-file-name id label)]))
```

Comment:
why did you change the storage path?

Reply:
Docstring wasn't clear enough that it lured me into thinking that the code is too complex. 😁